t/backup_expiration.t

```diff
@@ -0,0 +1,250 @@
+      it "succeeds" => sub {
+        my $exp = HTFeed::BackupExpiration->new(storage_name => 'prefixedversions-test');
+        ok($exp, 'new returns a value');
+      };
+    };
+
+    describe "#run" => sub {
+      it "does not do anything with a single version" => sub {
+        my $storage = staged_volume_storage('prefixedversions-test', 'test','test');
+        my $version = old_random_timestamp();
+        $storage->{timestamp} = $version;
+        $storage->stage;
+        $storage->make_object_path;
+        $storage->move;
+        $storage->record_backup;
+        my $exp = HTFeed::BackupExpiration->new(storage_name => 'prefixedversions-test');
+        $exp->run();
+        my $sql = 'SELECT COUNT(*) FROM feed_backups' .
+                  ' WHERE namespace=? AND id=? AND version=? AND deleted=1';
+        my @res = get_dbh->selectrow_array($sql, undef, 'test', 'test', $version );
+        is($res[0], 0, 'object is not deleted');
+        my $mets = $storage->mets_obj_path();
+        my $zip = $storage->zip_obj_path();
+        ok(-e $mets, "new ($version) mets left intact");
+        ok(-e $zip, "new ($version) zip left intact");
+      };
+
+      it "deletes old versions when there is a new one" => sub {
+        my @old_versions;
+        foreach my $n (1 .. 2) {
+          my $storage = staged_volume_storage('prefixedversions-test', 'test','test');
+          push @old_versions, $storage;
+          my $version = old_random_timestamp();
+          $storage->{timestamp} = $version;
+          $storage->stage;
+          $storage->make_object_path;
+          $storage->move;
+          $storage->record_backup;
+        }
+        my $storage = staged_volume_storage('prefixedversions-test', 'test','test');
+        my $new_version = new_random_timestamp();
+        $storage->{timestamp} = $new_version;
+        $storage->stage;
+        $storage->make_object_path;
+        $storage->move;
+        $storage->record_backup;
+        my $exp = HTFeed::BackupExpiration->new(storage_name => 'prefixedversions-test');
+        $exp->run();
+        my $sql = 'SELECT COUNT(*) FROM feed_backups' .
+                  ' WHERE namespace=? AND id=? AND deleted=1';
+        my @res = get_dbh->selectrow_array($sql, undef, 'test', 'test');
+        is($res[0], 2, 'two old objects marked with feed_backups.deleted=1');
+        foreach my $old_storage (@old_versions) {
+          my $mets = $old_storage->mets_obj_path();
+          my $zip = $old_storage->zip_obj_path();
+          ok(!-e $mets, "old ($old_storage->{timestamp}) mets deleted");
+          ok(!-e $zip, "old ($old_storage->{timestamp}) zip deleted");
+        }
+        my $mets = $storage->mets_obj_path();
+        my $zip = $storage->zip_obj_path();
+        ok(-e $mets, "new ($new_version) mets left intact");
+        ok(-e $zip, "new ($new_version) zip left intact");
+      };
+
+      it "does not do anything with old versions if there is no new one" => sub {
```

Comment:
I'm not sure this (and the one at line 223) are quite right - in this scenario, I would expect this could happen if we hadn't run the expiration/cleanup script recently, and the expected behavior would be that the oldest version was pruned.

I think it's also worth testing what happens if we have two versions, *neither* of which is older than 6 months -- in this case it should leave both new versions intact.